streamclient: panic in TestSinklessReplicationClient

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/3234996?expandedTest=build%3A%28id%3A3234990%29%2Cid%3A130957

```
panic: close of closed channel

goroutine 2276 [running]:
github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient.(*sinklessReplicationClient).ConsumePartition.func1(0xc0003d6540, 0xc0022c47e0, 0xc00136c9c0, 0xc0028c1200, 0x58448f0, 0xc00218ce00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient/cockroach_sinkless_replication_client.go:128 +0x6f5
created by github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient.(*sinklessReplicationClient).ConsumePartition
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient/cockroach_sinkless_replication_client.go:88 +0x3f0

goroutine 1 [chan receive]:
testing.(*T).Run(0xc000489680, 0x4468a5f, 0x1d, 0x4a73ca0, 0x4b31a6)
	/usr/local/go/src/testing/testing.go:1239 +0x2da
testing.runTests.func1(0xc000489500)
	/usr/local/go/src/testing/testing.go:1511 +0x78
testing.tRunner(0xc000489500, 0xc000c7fd60)
	/usr/local/go/src/testing/testing.go:1193 +0xef
testing.runTests(0xc0007da060, 0x75cb490, 0x1, 0x1, 0xc03869a2bab3f627, 0x274afbd7e58, 0x7931660, 0xc000978740)
	/usr/local/go/src/testing/testing.go:1509 +0x2fe
testing.(*M).Run(0xc000bc7d00, 0x0)
	/usr/local/go/src/testing/testing.go:1417 +0x1eb
github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient.TestMain(0xc000bc7d00)
	/go/src/github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamclient/main_test.go:31 +0xde
main.main()
	_testmain.go:47 +0x165
```

--- a/pkg/ccl/streamingccl/streamclient/BUILD.bazel
+++ b/pkg/ccl/streamingccl/streamclient/BUILD.bazel
@@ -53,7 +53,6 @@ go_test(
         "//pkg/server",
         "//pkg/sql/catalog/catalogkv",
         "//pkg/testutils/serverutils",
-        "//pkg/testutils/skip",
         "//pkg/testutils/testcluster",
         "//pkg/util/hlc",
         "//pkg/util/leaktest",

--- a/pkg/ccl/streamingccl/streamclient/cockroach_sinkless_replication_client_test.go
+++ b/pkg/ccl/streamingccl/streamclient/cockroach_sinkless_replication_client_test.go
@@ -18,7 +18,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamingtest"
 	_ "github.com/cockroachdb/cockroach/pkg/ccl/streamingccl/streamproducer" // Ensure we can start replication stream.
 	"github.com/cockroachdb/cockroach/pkg/sql/catalog/catalogkv"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/stretchr/testify/require"
@@ -27,9 +26,10 @@ import (
 // channelFeedSource wraps the eventsCh returned from a client. It expects that
 // no errors are returned from the client.
 type channelFeedSource struct {
-	t       *testing.T
-	eventCh chan streamingccl.Event
-	errCh   chan error
+	t               *testing.T
+	cancelIngestion context.CancelFunc
+	eventCh         chan streamingccl.Event
+	errCh           chan error
 }
 
 var _ streamingtest.FeedSource = (*channelFeedSource)(nil)
@@ -50,14 +50,11 @@ func (f *channelFeedSource) Next() (streamingccl.Event, bool) {
 
 // Close implements the streamingtest.FeedSource interface.
 func (f *channelFeedSource) Close() {
-	close(f.eventCh)
+	f.cancelIngestion()
 }
 
 func TestSinklessReplicationClient(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-
-	// channel usage error in this test
-	skip.WithIssue(t, 68168)
 
 	defer log.Scope(t).Close(t)
 	h, cleanup := streamingtest.NewReplicationHelper(t)
@@ -99,7 +96,7 @@ INSERT INTO d.t2 VALUES (2);
 		clientCtx, cancelIngestion := context.WithCancel(ctx)
 		eventCh, errCh, err := client.ConsumePartition(clientCtx, pa, startTime)
 		require.NoError(t, err)
-		feedSource := &channelFeedSource{eventCh: eventCh, errCh: errCh}
+		feedSource := &channelFeedSource{cancelIngestion: cancelIngestion, eventCh: eventCh, errCh: errCh}
 		feed := streamingtest.MakeReplicationFeed(t, feedSource)
 
 		// We should observe 2 versions of this key: one with ("привет", "world"), and a later

--- a/pkg/ccl/streamingccl/streamingtest/replication_helpers.go
+++ b/pkg/ccl/streamingccl/streamingtest/replication_helpers.go
@@ -98,7 +98,7 @@ func (rf *ReplicationFeed) Close() {
 }
 
 func (rf *ReplicationFeed) consumeUntil(pred FeedPredicate) error {
-	const maxWait = 10 * time.Second
+	const maxWait = 20 * time.Second
 	doneCh := make(chan struct{})
 	mu := struct {
 		syncutil.Mutex

